sql/analyzer/resolve_subqueries.go

```diff
@@ -50,9 +50,31 @@ func finalizeSubqueries(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope,
 // finalizeSubqueriesHelper finalizes all subqueries and subquery expressions,
 // fixing parent scopes before recursing into child nodes.
 func finalizeSubqueriesHelper(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
-	return transform.Node(node, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	var joinParent *plan.JoinNode
+	var selFunc transform.SelectorFunc = func(c transform.Context) bool {
+		if jp, ok := c.Node.(*plan.JoinNode); ok {
+			joinParent = jp
+		}
+		return true
+	}
+
+	var conFunc transform.CtxFunc = func(c transform.Context) (sql.Node, transform.TreeIdentity, error) {
+		n := c.Node
 		if sqa, ok := n.(*plan.SubqueryAlias); ok {
-			newSqa, same2, err := analyzeSubqueryAlias(ctx, a, sqa, scope, sel, true)
+			var newSqa sql.Node
+			var same2 transform.TreeIdentity
+			var err error
+			if sqa.OuterScopeVisibility && joinParent != nil && !joinParent.Op.IsLeftOuter() {
```

Comment:
`!joinParent.Op.IsLeftOuter()` seems suspicious, why would any of the indexing be dependent on the parent being a left join? We should probably add a test for left joins.

Reply:
The `joinParent.Op.IsLeftOuter()` check might've been added to prevent breaking other queries in a previous iteration, but it looks like it isn't needed now. I'll remove it.